### DISCUSSION TITLE
Integration with the Solid Project

### DISCUSSION THREAD

**Discussion by gsvarovsky** [m-ld/m-ld-spec#104] [Discussion]
[Solid Forum topic](https://forum.solidproject.org/t/live-shared-pod-data-making-it-happen/4061)

[m-ld discussions topic](https://github.com/m-ld/m-ld-spec/discussions/71)

Research, prototype, and test a technology for real-time group collaborative editing of structured web resources, securely, and respecting privacy.

Achieve this by combining Solid, a specification that lets people store their data securely in decentralised data stores, and m-ld, a technology for real-time information sharing.

These technologies share a common data representation, making the combination natural and harmonious. Solid will provide for Identity, Authentication, Authorisation and server storage of resources, while m-ld will provide real-time synchronisation of resource replicas, including locally on devices.

Benefits to users of apps built with this innovation would include the ability to collaborate on documents in real-time, the ability to work offline, and greater performance and resilience of the app, so that they are always able to continue acting on information.

[NGI Pointer - attachment.pdf](https://github.com/m-ld/m-ld-spec/files/6579923/NGI.Pointer.-.attachment.pdf)

**Comment by franzzua** [comment]
WOW, i'm just in searching for a CRDT library for use with solid-community server)
So, i would implement PubSub remote and publish it as a component for solid-community.
Is there any guidelines for implementing RubsubRemote?

**Comment by gsvarovsky** [comment]
Very interesting @franzzua! can you describe more what you are trying to achieve? (Note this repo has a [discussion board](https://github.com/m-ld/m-ld-spec/discussions) if you like).

Implementing a new `PubsubRemotes` is for using a new low-level messaging protocol, to publish messages between clones (for examples, see https://github.com/m-ld/m-ld-spec/issues/90). That might not be what you really want to do. What protocol are you thinking of using? There are no guidelines yet, but I'd be happy to help!

Since Solid is server-based, it might be best to start by using the socket.io remotes and have a socket.io service.

Or, are you thinking of using a Solid "inbox" for messaging? That might not be very fast/efficient, but, it all depends.

Also, the "architecture" section in the [attached document](https://github.com/m-ld/m-ld-spec/files/6579923/NGI.Pointer.-.attachment.pdf) might be helpful.

**Comment by gsvarovsky** [comment]
Discussion moved to #71
